src/mem/metaslab.h

```diff
@@ -163,14 +176,30 @@ namespace snmalloc
     {
       return meta &= ~META_BOUNDARY_BIT;
     }
+    ///@}
 
+    /**
+     * Returns the remote.
+     *
+     * If the meta entry is owned by the back end then this returns an
+     * undefined value and will abort in debug builds.
+     */
     [[nodiscard]] SNMALLOC_FAST_PATH RemoteAllocator* get_remote() const
     {
+      SNMALLOC_ASSERT(!is_backend_owned());
       return reinterpret_cast<RemoteAllocator*>(
         pointer_align_down<REMOTE_WITH_BACKEND_MARKER_ALIGN>(
           get_remote_and_sizeclass()));
     }
 
+    /**
+     * Return the sizeclass.
+     *
+     * This is called by `external_pointer` on arbitrary memory and so cannot
+     * assert that the meta entry is owned by the front end.  In the future, it
+     * may provide some stronger guarantees on the value that is returned in
+     * this case.
+     */
     [[nodiscard]] SNMALLOC_FAST_PATH sizeclass_t get_sizeclass() const
     {
```

Comment:
```suggestion
    {
      SNMALLOC_ASSERT(!is_backend_owned());
```

Reply:
See the comment on this method for an explanation of why this method does not contain that assert.

Reply:
Sorry, missed that.  Can we assert, then, that it's in one of three non-overlapping states: unowned, owned by backend and sizeclass zero, or owned by frontend?  Or do we not care if we generate nonsensical results from `external_pointer` for things that are backend owned if we allow the backend to reuse the sizeclass bits?

Reply:
That assert would check that one of the following holds:

 - All bits are zero.
 - All bits except bit N are in an undefined state, bit N is 1.
 - All bits except bit N are in an undefined state, bit N is 0.

For any bit pattern, exactly one out of conditions 2 or 3 will always hold.